includes/api/class-plugins-controller.php

```diff
@@ -0,0 +1,463 @@
+<?php
+/**
+ * Plugin management API endpoints.
+ *
+ * @package Newspack\API
+ */
+
+namespace Newspack\API;
+
+use \WP_REST_Controller;
+use \WP_Error;
+use Newspack\Plugin_Manager;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST API endpoints for managing plugins.
+ */
+class Plugins_Controller extends WP_REST_Controller {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = '/newspack/v1';
+
+	/**
+	 * Endpoint resource.
+	 *
+	 * @var string
+	 */
+	protected $resource_name = 'plugins';
+
+	/**
+	 * Register the routes.
+	 */
+	public function register_routes() {
+		// Register newspack/v1/plugins endpoint.
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->resource_name,
+			[
+				[
+					'methods'             => 'GET',
+					'callback'            => [ $this, 'get_items' ],
+					'permission_callback' => [ $this, 'get_items_permissions_check' ],
+				],
+				'schema' => [ $this, 'get_item_schema' ],
+			]
+		);
+
+		// Register newspack/v1/plugins/some-plugin endpoint.
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->resource_name . '/(?P<slug>[\a-zA-Z]+)',
```

Comment:
Curious if we need to support upper and lower case plugin slugs? I was under the impression they are all lowercase although I could be wrong about that?

Reply:
They all get sanitized to lowercase before processing, so it doesn't matter much. I suppose there's no reason to allow uppercase.

Reply:
I've removed support for uppercase in slugs.